mne/beamformer/_dics.py

```diff
@@ -1,113 +1,440 @@
             raise ValueError('data and picks must have the same length')
 
-        if not return_single:
+        if not one_epoch:
             logger.info("Processing epoch : %d" % (i + 1))
 
         # Apply SSPs
         if info['projs']:
-            M = np.dot(proj, M)
+            _check_proj_match(info, filters)
+            M = np.dot(filters['proj'], M)
 
-        # project to source space using beamformer weights
-        if is_free_ori:
+        stcs = []
+        for W in Ws:
+            # project to source space using beamformer weights
             sol = np.dot(W, M)
-            logger.info('combining the current components...')
-            sol = combine_xyz(sol)
+
+            if filters['n_orient'] > 1:
+                logger.info('combining the current components...')
+                sol = combine_xyz(sol)
+
+            tstep = 1.0 / info['sfreq']
+
+            # XXX : STC cannot contain (yet?) complex values
+            if np.iscomplexobj(sol):
+                sol = np.abs(sol)
+            stcs.append(_make_stc(sol, vertices=filters['vertices'], tmin=tmin,
+                                  tstep=tstep, subject=subject))
+        if one_freq:
+            yield stcs[0]
         else:
-            # Linear inverse: do not delay compuation due to non-linear abs
-            sol = np.dot(W, M)
+            yield stcs
+
+    logger.info('[done]')
+
+
+@verbose
+def apply_dics(evoked, filters, verbose=None):
+    """Apply Dynamic Imaging of Coherent Sources (DICS) beamformer weights.
+
+    Apply Dynamic Imaging of Coherent Sources (DICS) beamformer weights
+    on evoked data.
+
+    .. warning:: The result of this function is meant as an intermediate step
+                 for further processing (such as computing connectivity). If
+                 you are interested in estimating source time courses, use an
+                 LCMV beamformer instead. If you are interested in estimating
+                 spectral power at the source level, use
+                 :func:`dics_source_power`.
+    .. warning:: This implementation has not been heavily tested so please
+                 report any issues or suggestions.
+
+    Parameters
+    ----------
+    evoked : Evoked
+        Evoked data to apply the DICS beamformer weights to.
+    filters : dict
+        DICS spatial filter (beamformer weights)
+        Filter weights returned from :func:`make_dics`.
+    verbose : bool | str | int | None
+        If not `None`, override default verbose level (see :func:`mne.verbose`
+        and :ref:`Logging documentation <tut_logging>` for more).
+
+    Returns
+    -------
+    stc : SourceEstimate | VolSourceEstimate | list of (SourceEstimate | VolSourceEstimate)
+        Source time courses. If the DICS beamformer has been computed for more
+        than one frequency, a list is returned containing for each frequency
+        the corresponding time courses.
+
+    See Also
+    --------
+    apply_dics_epochs
```

Comment:
should also have at least `dics_source_power`

Reply:
I added a link to `apply_dics_csd`